packages/modules/product/src/repositories/product.ts

```diff
@@ -13,28 +17,78 @@ export class ProductRepository extends DALUtils.mikroOrmBaseRepositoryFactory(
     super(...arguments)
   }
 
+  /**
+   * Identify the relations to load for the given update.
+   * @param update
+   * @returns
+   */
+  getRelationsToLoad(update: any[]): string[] {
+    const relationsToLoad = new Set<string>()
+    update.forEach((update) => {
+      if (update.options) {
+        relationsToLoad.add("options")
+        relationsToLoad.add("options.values")
+      }
+      if (update.variants) {
+        relationsToLoad.add("options")
+        relationsToLoad.add("options.values")
+        relationsToLoad.add("variants")
+        relationsToLoad.add("variants.options")
+        relationsToLoad.add("variants.options.option")
+      }
+      if (update.tags) relationsToLoad.add("tags")
```

Comment:
Would it make sense to have an `update.meta` property that stores "options", "variants"... and segregate cases where `update.meta === 'variants'` and for the rest just add `update.meta` to the `relationsToLoad` set?

Reply:
just saw your comment, the pr is still in progress 😅 
This suggestion would require each product to be fetch independently for the relations to be determined by product to update right?